numpy/core/numeric.py

```diff
@@ -1527,15 +1527,12 @@ def rollaxis(a, axis, start=0):
 
     """
     n = a.ndim
-    if axis < 0:
-        axis += n
+    axis = resolve_axis(axis, n)
     if start < 0:
         start += n
```

Comment:
This has always struck me as a bug - IMO, `rollaxis(np.zeros(3,4,5), 0, -1).shape == (4, 5, 3)`, ie `-1` should mean "move this axis to the end. The fact that this is the only place that can't use `resolve_axis` sort of backs that up.